feat: Docker compose for local development w no hassle

xcfw opened the pull request:
# Description

> Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

> Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

> Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:

- OS with version:
- Browser with version:
- SDK / CLI with version:

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I documented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

xcfw commented:
@dahal I fixed the problem with watch mode of nextjs when running using compose and rewrote compose file and all dockerfiles for local development, also added notes to documentation. Please review.

dahal commented:
> @dahal I fixed the problem with watch mode of nextjs when running using compose and rewrote compose file and all dockerfiles for local development, also added notes to documentation. Please review.

@xcfw thanks for submitting this PR. Its really shaping up. However, I am running into this issue when running 

`docker compose up platform --build`
![Screen Shot 2023-06-07 at 12 22 04 AM](https://github.com/envless/envless/assets/3684236/c8b0e055-314a-4471-b2c2-51725529f5a4)

xcfw commented:
> > @dahal I fixed the problem with watch mode of nextjs when running using compose and rewrote compose file and all dockerfiles for local development, also added notes to documentation. Please review.
> 
> @xcfw thanks for submitting this PR. Its really shaping up. However, I am running into this issue when running
> 
> `docker compose up platform --build` ![Screen Shot 2023-06-07 at 12 22 04 AM](https://user-images.githubusercontent.com/3684236/243921310-c8b0e055-314a-4471-b2c2-51725529f5a4.png)

This is fixed as well, pls see the [last commit](https://github.com/envless/envless/pull/291/commits/a7e11cbad2959be7a1f43d2ab1863e85cd1fb1d2)

xcfw commented on package.json:
```diff
@@ -17,7 +17,7 @@
     "test:e2e": "dotenv -- turbo run test:e2e",
     "db:generate": "dotenv -- turbo run db:generate",
-    "db:migrate": "dotenv -- yarn --cwd ./apps/platform/ prisma migrate dev",
+    "db:migrate": "dotenv -- yarn --cwd /platform/ prisma migrate dev",
```
For those who can not develop with docker they need to make sure that yarn cwd argument is set to `--cwd ./apps/platform/`

dahal replied:
Good catch 💙. Let's create a different command for this in that case. Asking users to make those changes is not ideal.

`docker:db:migrate`

dahal replied:
I have pushed a fix for this. ✅

dahal reviewed (approved):
Amazing work on this one @xcfw 👏🏼
